src/components/loginRegister/Register.jsx

```diff
@@ -142,15 +142,28 @@ const Register = () => {
         });
       }
     } catch (error) {
-      // Registration failed
-      notification.error({
-        message: "Registration Failed",
-        description: "An error occurred during registration. Please try again.",
-        icon: <ExclamationCircleTwoTone twoToneColor="#eb2f96" />,
-        placement: "top",
-        duration: 4.5,
-        style: { width: 300 },
-      });
+      if (error.status === 400) {
+        // User already exists
+        notification.error({
```

Comment:
So if you're other function throws an error something like:
```
throw new Error('registration failed because a user with that email already exists')
```

You'd maybe want to have the code in here like this:
```js
try {
  // request is sent here
} catch(err) {
  if (err.message.includes('email already exists')) {
     // show message about email already existing
  } 

  // show generic error message
}
```

Reply:
How come you are always using `.includes` instead of requiring the error message to match something you define? 

or for `response_body.detail === "account with that email already exists"`

instead of `response_body.detail.includes('email already exists')`

In register.jsx I have 
![image](https://github.com/rknizzle/propermesh-view/assets/111093132/cb0919f6-1c78-4d9f-89c0-814d7953b47c)

which would match the error thrown in registerRequest.js

![image](https://github.com/rknizzle/propermesh-view/assets/111093132/407c448d-9803-4561-b080-796ed3a63269)

Reply:
Is this incase you change the response_body.detail in the future? and it will most likely still contain `email already exists`?

Reply:
Yea exactly; either way will work I just feel like this method gives a little more flexibility to maybe still work if the error message on the backend changes slightly. A change to the message on the backend would realistically be very likely to make this fail anyway though so it doesnt really matter